### PR TITLE
Add docker

### PR DESCRIPTION
**Pull request by joshsmith**
This adds Docker to replace Vagrant, since that's so yesterday. Literally.

I still need to rewrite docs to account for this, but it appears to be working. Pushing up so we can test locally some.

### CODE REVIEW

File: bin/setup [hunk 1]
```diff
@@ -3,41 +3,37 @@
 # Exit if any subcommand fails
 set -e
 
+# Import colors
+. colors
```
Comment (npendery) [inline bin/setup]: Got an error with this. I changed it to be `. bin/colors` and it worked

File: bin/setup [hunk 2]
```diff
@@ -3,41 +3,37 @@
 # Exit if any subcommand fails
 set -e
 
+# Import colors
+. bi/colors
```
Comment (christopherstyles) [inline bin/setup]: Missing an “n” here I believe (`. bin/colors`).

Comment (joshsmith) [inline bin/setup]: Fixed.

File: Guardfile [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+# Note: The cmd option is now required due to the increasing number of ways
+#       rspec may be run, below are examples of the most common uses.
+#  * bundler: 'bundle exec rspec'
+#  * bundler binstubs: 'bin/rspec'
+#  * spring: 'bin/rspec' (This will use spring if running and you have
+#                          installed the spring binstubs per the docs)
+#  * zeus: 'zeus rspec' (requires the server to be started separately)
+#  * 'just' rspec: 'rspec'
+
+guard :rspec, cmd: "bundle exec rspec" do
```
Comment (christopherstyles) [inline Guardfile:27]: I’d suggest adding a spring binstub for rspec with `bundle exec spring binstub rspec` then changing this line to:

``` ruby
guard :rspec, cmd: "bin/rspec" do
```

In my tests it was trimming 7–9 seconds off load time.

**Comment by christopherstyles** [comment]
👍 :shipit:

**Pull request merged**